Source/AwsCommonRuntimeKit/http/HTTPStreamCallbackCore.swift

```diff
@@ -70,13 +70,16 @@ private func onResponseHeaderBlockDone(stream: UnsafeMutablePointer<aws_http_str
             """
         )
     }
-    switch HTTPHeaderBlock(rawValue: headerBlock) {
-    case .informational:
-        httpStreamCbData.requestOptions.onInterimResponse?(UInt32(status), httpStreamCbData.headers)
-    case .main:
-        httpStreamCbData.requestOptions.onResponse(UInt32(status), httpStreamCbData.headers)
-    case .trailing:
-        httpStreamCbData.requestOptions.onTrailer?(httpStreamCbData.headers)
+
+    if let headerBlock = HTTPHeaderBlock(rawValue: headerBlock) {
+        switch headerBlock {
+        case .informational:
+            httpStreamCbData.requestOptions.onInterimResponse?(UInt32(status), httpStreamCbData.headers)
+        case .main:
+            httpStreamCbData.requestOptions.onResponse(UInt32(status), httpStreamCbData.headers)
+        case .trailing:
+            httpStreamCbData.requestOptions.onTrailer?(httpStreamCbData.headers)
+        }
     }
```

Comment:
If the failable `HTTPHeaderBlock` initializer returns `nil`, this switch "fails silently" (i.e. does nothing.). Is that desired?

Reply:
Thanks, fixed by `!` as we don't expect this enum to have unknown types.